### ISSUE TITLE
Upgrading within the same plan is now fixed in terms of actually updating both the primary and secondary DB instance, but still seeing issues upgrading from Basic to Standard tier with the latest release (1.0.0-rc39):

### ISSUE THREAD

**Issue by blgm** [cloudfoundry-incubator/csb-brokerpak-azure#47]
Upgrading within the same plan is now fixed in terms of actually updating both the primary and secondary DB instance, but still seeing issues upgrading from Basic to Standard tier with the latest release (1.0.0-rc39):

```
status:    update failed
message:   Error: waiting for creation of MsSql Database "csb-fog-db-bf834bab-cca9-4c3d-bcc6-1370cd561773" (MsSql Server Name "azugessqlsandbox" / Resource Group "MFC-USE-PCF-PSB"): Code="SourceDatabaseEditionCouldNotBeUpgraded" Message="The source database
           'azugessqlsandbox.csb-fog-db-bf834bab-cca9-4c3d-bcc6-1370cd561773' cannot have higher edition than the target database 'azugessqlsandboxdr.csb-fog-db-bf834bab-cca9-4c3d-bcc6-1370cd561773'. Upgrade the edition on the target before upgrading source."  on main.tf
           line 15, in resource "azurerm_mssql_database" "primary_db":  15: resource "azurerm_mssql_database" "primary_db" { exit status 1
```

_Originally posted by @claassen in https://github.com/cloudfoundry-incubator/csb-brokerpak-azure/issues/20#issuecomment-849804451_

**Conversation went:**
@aristosvo:
> Yeah, tested that as well and didn't implement the solution yet, as it's not clear whether Azure supports it: https://docs.microsoft.com/en-us/azure/azure-sql/database/active-geo-replication-overview#upgrading-or-downgrading-primary-database.

> If you have a working example how to do it with az cli, I might spend some time on implementing it.

@claassen:
> @aristosvo I'm not familiar with the internal workings of Terraform to know if how feasible this is or not, but it seems like the solution should be as simple as just ensuring the secondary DB is updated before the primary. Our workaround has been to manually update the secondary DB in Azure, followed by the primary in order to do plan updates across tiers.

@aristosvo:
> Okay, looks like I have a solution here, but I need to test this extensively as this may contain a few edge cases.

> I'm especially not experienced with elastic pools, so if you have experience scaling from single database to pools in FOG, let me know.

**Comment by blgm** [comment]
@aristosvo [you mentioned](https://github.com/cloudfoundry-incubator/csb-brokerpak-azure/issues/20#issuecomment-850209145) that you had a solution that you were testing. Is there anything that we can do to help out?

**Comment by aristosvo** [comment]
@blgm Do you have a time machine by any chance? 🤣

I'm afraid I lost it somehow, I'm very sorry. I've completed multiple PR's in the meantime, must've got rid of the changes somehow.. 😢

**Comment by pivotal-marcela-campo** [comment]
Hi @claassen, Could you help us reproduce this issue? 
In particular, how are the basic and standard plans you are using configured? 

I am trying this

```
>>  cf create-service csb-azure-mssql-failover-group small-v2 foggy-root -b csb-root
>>  cf update-service foggy-root -p large
```

And it works fine. As our plans are based on a different pricing model I am having problems figuring out what plans I should create to reproduce this issue.

I would need to reproduce to raise a bug on the terraform provider side for it to be tackled.
Thanks!

**Comment by claassen** [comment]
I think the difference might be that we are using DTU based plans. Here's the config we are using, updating from Basic to StandardS0 is what fails for us:

```
{
        "name": "Basic",
        "id": "d219369e-9a0d-481b-b6ae-dc57ea5f204e",
        "description": "Basic DTU based plan",
        "sku_name": "Basic",
        "max_storage_gb": 2
      },
      {
        "name": "StandardS0",
        "id": "535fd89e-4514-4bae-8e3d-fe85db6ddfdb",
        "description": "StandardS0 DTU based plan",
        "sku_name": "S0",
        "max_storage_gb": 250
      },
```

**Comment by pivotal-marcela-campo** [comment]
Thanks @claassen 

Created [this bug](https://github.com/terraform-providers/terraform-provider-azurerm/issues/12235) on the terraform provider to get this fixed.

@aristosvo tagging you just in case you are interested in taking a look again.

**Comment by claassen** [comment]
Just a heads up it looks like this should now be fixed in version 2.79.0 of the azurerm provider.

**Comment by FelisiaM** [comment]
Hi @claassen 

We tested with the latest Terraform provider and it looks like it has solved the issue 🎉
We have now released the latest brokerpak [v1.1.0-rc2](https://github.com/cloudfoundry-incubator/csb-brokerpak-azure/releases/tag/1.1.0-rc.2) including the latest provider.
